Libs/MRML/Core/vtkMRMLBitStreamVolumeNode.cxx

```diff
@@ -0,0 +1,247 @@
+    {
+    this->CompressionDevice->SetContentImage(this->GetImageData());
+    }
+  std::string compressedStream = this->CompressionDevice->GetCompressedBitStreamFromData();
+  if (compressedStream.size()>0)
+    {
+    this->SetFrameMessage(compressedStream);
+    if(!this->GetKeyFrameReceived())
+      {
+      this->SetKeyFrameReceived(true);
+      this->SetKeyFrameMessage(compressedStream);
+      }
+      return 1;
+    }
+  return 0;
+}
+
+std::string vtkMRMLBitStreamVolumeNode::GetCodecType()
+{
+  vtkMRMLCompressionDeviceNode*  device = this->GetCompressionDevice();
+  if (device)
+    {
+    return device->GetCurrentCodecType();
+    }
+  return "";
+}
+
+
+int vtkMRMLBitStreamVolumeNode::SetCodecType(std::string name)
+{
+  vtkMRMLCompressionDeviceNode*  device = this->GetCompressionDevice();
+  if(device)
+  {
+    device->SetCurrentCodecType(name);
+    return 1;
+  }
+  return 0;
+}
+
+void vtkMRMLBitStreamVolumeNode::ProcessMRMLEvents(vtkObject *caller, unsigned long event, void *callData )
+{
+  this->vtkMRMLNode::ProcessMRMLEvents(caller, event, callData);
+  this->InvokeEvent(ImageDataModifiedEvent);
+}
+
+void vtkMRMLBitStreamVolumeNode::ProcessDeviceModifiedEvents( vtkObject *caller, unsigned long event, void *callData )
+{
+  vtkMRMLCompressionDeviceNode* modifiedDevice = reinterpret_cast<vtkMRMLCompressionDeviceNode*>(caller);
+  if (modifiedDevice == NULL)
+    {
+    // we are only interested in proxy node modified events
+    return;
+    }
+  if (event != modifiedDevice->GetDeviceContentModifiedEvent())
+    {
+    return;
+    }
+  long streamLength = modifiedDevice->GetContent()->frameMessage.length();
+  std::string* bitStream = new std::string(modifiedDevice->GetContent()->frameMessage);
+  this->SetFrameUpdated(true);
+  this->SetKeyFrameUpdated(false);
+  this->SetFrameMessage(*bitStream);
+  if(!this->GetKeyFrameReceived() || modifiedDevice->GetContent()->keyFrameUpdated)
+    {
+    streamLength = modifiedDevice->GetContent()->keyFrameMessage.length();
+    std::string*  keyFrameStream = new std::string(modifiedDevice->GetContent()->keyFrameMessage);
+    this->SetKeyFrameMessage(*keyFrameStream);
+    this->SetKeyFrameReceived(true);
+    this->SetKeyFrameUpdated(true);
+    delete keyFrameStream;
+    }
+  delete bitStream;
+  this->Modified();
+}
+
+//----------------------------------------------------------------------------
+vtkMRMLCompressionDeviceNode* vtkMRMLBitStreamVolumeNode::GetCompressionDevice()
+{
+  return this->CompressionDevice;
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLBitStreamVolumeNode::WriteXML(ostream& of, int nIndent)
+{
+  Superclass::WriteXML(of, nIndent);
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLBitStreamVolumeNode::ReadXMLAttributes(const char** atts)
+{
+  int disabledModify = this->StartModify();
+  
+  Superclass::ReadXMLAttributes(atts);
```

Comment:
Codec name must be read/written/printed/copied. Use the recently added helper macros https://github.com/Slicer/Slicer/blob/master/Libs/MRML/Core/vtkMRMLCameraNode.cxx#L110

Reply:
Addressed